Refactor Line Merging of Equations, on Backspace.

matthewborden opened the pull request:
When there is an equation at the end of a line and we are backspacing
from the line below. Instead of trying to treat the equation at the end
of a line as a special case (and delete it), we place the cursor at the end of the
equation and then let it be handled as the case of deleting a equation
within a line, thus the user presses backspace once to go up a line and 
then again to delete an equation.

Commit:
Refactor Line Merging of multiple lines.

When there is an equation at the end of a line and we are backspacing
from the line below. Instead of trying to treat the equation at the end
of a line as a special case, we place the cursor at the end of the
equation and then let it be handled as the case of deleting a equation
within a line.

Commit:
Use variable reference to ancestor, instead of direct reference.

matthewborden commented on build/squire-raw.js:
```diff
@@ -3364,7 +3342,28 @@ var keyHandlers = {
+                    element we treat it as an element that is a descendant of a <svg>. We loop up the tree until we find an 
+                    element with the mathjax class and then use that to set the caret after the equation.
+                */
+                if (ancestor.querySelector && !ancestor.querySelector("mathjax") && checkInSVG.test(ancestor)) {
```
Should test ancestor.nodeName

djrodgerspryor commented:
The changes look good, but can we not commit the build files?

Just add build/ to .gitignore

Commit:
Test against nodeName instead of element.

Minor fix: test against ancestor.nodeName instead of ancestor.

matthewborden commented:
Just to clarify, build/squire.js is the file used in production.

We could use https://www.npmjs.com/package/gulp-exec to run `make` during the web client build process, that would generate the build directory. Another option would be a githook that reminds us to run `make` before committing and then keep the build directory in the repo. 

Note: The Squire project currently commits the build directory. https://github.com/neilj/Squire

Commit:
Remove Build Directory from Squire Repo.
Run make clean && make instead, from the web-client build file (gulpfile.js), instead of commiting the static files. This is due to the fact that someone will make a change to the source folder and then forget to run make, thus their changes won't have any effect.

matthewborden commented:
Closing until changes to the build process are merged.
